ykrt/src/compile/jitc_yk/aot_ir.rs

```diff
@@ -743,6 +748,7 @@ impl IntegerType {
     /// Create a new integer type with the specified number of bits.
     #[cfg(test)]
     pub(crate) fn new(num_bits: u32) -> Self {
+        debug_assert!(num_bits > 0 && num_bits < 0x7fffff);
```

Comment:
Is the upper bound of 2^23 inclusive? LLVM's docs aren't clear.

To answer this:
```
$ cat a.ll  
define void @f(i8388608) {
  ret void
}
$ clang a.ll
... compiles forever
```

but if you add one more bit to the integer type, `i8388609`, then:
```
$ clang a.ll
a.ll:1:16: error: expected type
define void @f(i8388609) {
               ^
1 error generated.
```

So the upper bound check should be `num_bits <= 0x800000`, right?

Reply:
Good catch!

Reply:
Fixed in https://github.com/ykjit/yk/pull/1075/commits/07fad1825e1ceca905d3846a37bb27d1c2af4f98. I *think* that's everything now?

Reply:
Looks like there's still one 0x7fffff left? Or github's web ui is lying to me?

Reply:
I'm an idiot. Fixed in b6d49a9d36cf886275c90a95fd7e436f385d4599.

Reply:
NP, thanks